control-plane/agents/src/bin/core/controller/resources/resource_map.rs

```diff
@@ -91,4 +91,38 @@ where
             .map(|v| v.lock().clone())
             .collect()
     }
+
+    /// Apply the closure on the elements and return a subset of the map (i.e. a paginated result),
+    /// containing the elements for which closure returns true.
+    pub(crate) fn paginate_selected<F>(
+        &self,
+        max_entries: u64,
```

Comment:
why not reuse existing offset/len as the previous paginate?
Then simply filter from closure and apply same skip/take? You might even be able to rewrite things so both paginate functions reuse same code.

Reply:
Have modified to reuse existing paginate code as discussed. Pretty sleek !